assets/js/components/TableOverflowContainer.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * TableOverflowContainer component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { debounce } from 'lodash';
+import classnames from 'classnames';
+
+function TableOverflowContainer( { children } ) {
+	const [ isScrolling, setScrolling ] = useState( false );
+	const scrollRef = useRef();
+
+	useEffect( () => {
+		updateFadeOnScroll();
+
+		const resize = debounce( function() {
+			updateFadeOnScroll();
+		}, 100 );
+
+		global.addEventListener( 'resize', resize );
+
+		return () => global.removeEventListener( 'resize', resize );
+	}, [] );
+
+	const updateFadeOnScroll = () => {
+		const { scrollLeft, scrollWidth, offsetWidth } = scrollRef.current;
```

Comment:
It's possibly for `scrollRef.current` to be `null`, isn't it? I suppose it's unlikely given the logic flow of this being called in a `useEffect` hook, but I think it might be best to protect against `scrollRef.current` being null, just-in-case.